lib/resources/git_resource.py

```diff
@@ -27,90 +37,181 @@ def __init__(self, database: DbConnector):
         self._database = database
 
     @staticmethod
-    def clone_repository(technology, url, path):
+    def clone_repository(url, path) -> Repo:
         """
         This method tries to clone the repository from the provided url.
         """
         logger.debug(f"Cloning repository {url} ...")
-        Git(path).clone(url)
-        repository = Repo(f"{path}/{technology}")
-        return repository
+        return Repo.clone_from(url, path, bare=True)
 
     @staticmethod
-    def get_tags(repository: Repo):
+    def get_hash(file_content) -> str or None:
         """
-        This method returns the list of tags sorted by commit date.
+        This method computes the hash of the provided file and returns it.
         """
-        return sorted(repository.tags, key=lambda tag: tag.commit.committed_datetime)
+        if file_content is None:
+            return None
+        return hashlib.sha256(file_content).hexdigest()
 
     @staticmethod
-    def get_hash(file):
+    def get_all_files_from_commit(commit: Commit) -> List[Tuple[FilePath, BlobHash]]:
         """
-        This method computes the hash of the provided file and returns it.
+        This method retrieves all files with their blob hash in a commit.
+        """
+        file_list = []
+
+        for blob in commit.tree.traverse():
+            if S_ISDIR(blob.mode):
+                continue
+            file_list.append((blob.path, blob.hexsha))
+        return file_list
+
+    def _hash_files(
+        self,
+        files: List[Tuple[FilePath, TagName, BlobHash]],
+        repo_dir_path: str
+    ) -> List[Tuple[FilePath, TagName, FileHash]]:
+        """
+        This method calculates the SHA256 hashes of input files.
+        To do so, it reads first the content by using the file blob hash with the command `git cat-file -p [blob hash]`,
+        then it calculates the hash.
+        """
+        files_info: List[Tuple[FilePath, TagName, FilePath]] = []
+        current_dir = os.getcwd()
+        os.chdir(repo_dir_path)
+
+        for (file_path, tag_name, blob_hash) in files:
+            try:
+                file_content = subprocess.check_output(
```

Comment:
Here you could add a comment explaining why subprocess is the right choice and why you don't use the Gitpython way of interacting with the command line.